python/chemiscope/explore.py

```diff
@@ -0,0 +1,145 @@
+    :param callable featurize: optional. Function to compute features and perform
+        dimensionality reduction on the `frames`. The function should take `frames` as
+        input and return a 2D array of reduced features. If `None`, a default SOAP and
+        PCA based featurizer is used.
+
+    :param dict properties: optional. Additional properties to be included in the
+        visualization. This dictionary can contain any other relevant data associated
+        with the atomic structures. Properties can be extracted from frames with
+        :py:func:`extract_properties` or manually defined by the user.
+
+    :param str mode: optional. Visualization mode for the chemiscope widget. Can be one
+        of "default", "structure", or "map". The default mode is "default".
+
+    :return: a chemiscope widget for interactive visualization
+
+    Here is an example of usage with and without providing a function to do the
+    reprensentation and reduction. The frames are obtained by reading the structures
+    from a file that `ase <ase-io_>`_ can read, and performing PCA using `sklearn`_ on
+    a descriptor computed with SOAP with `dscribe`_ library.
+
+    .. code-block:: python
+
+        import chemiscope
+        from ase.io import read
+        from dscribe.descriptors import SOAP
+        from sklearn.decomposition import KernelPCA
+
+        # Read the structures from the dataset
+        frames = read("trajectory.xyz", ":")
+
+        # Basic usage with the default featurizer (SOAP + PCA)
+        chemiscope.explore(frames)
+
+        # Using a custom featurizer
+        chemiscope.explore(frames, featurize=soap_kpca)
+
+
+        def soap_kpca(frames):
+            # Compute descriptors
+            soap = SOAP(
+                species=["C"],
+                r_cut=4.5,
+                n_max=8,
+                l_max=6,
+                periodic=True,
+            )
+            descriptors = soap.create(frames)
+
+            # Apply KPCA
+            transformer = KernelPCA(n_components=2, gamma=0.05)
+
+            # Return a 2D array of reduced features
+            return transformer.fit_transform(descriptors)
+
+    .. _ase-io: https://wiki.fysik.dtu.dk/ase/ase/io/io.html
+    .. _sklearn: https://scikit-learn.org/
+    .. _dscribe: https://singroup.github.io/dscribe/latest/
+    """
+    # Validate inputs
+    if featurize is not None and not callable(featurize):
+        raise TypeError("'featurize' must be a callable (function)")
+    if properties is None:
+        properties = {}
+
+    # Apply dimensionality reduction from the provided featurizer
+    if featurize is not None:
+        X_reduced = featurize(frames)
+
+    # Use default featurizer
+    else:
+        X_reduced = soap_pca(frames)
+
+    # Add dimensionality reduction results to properties
+    settings = {"map": {}}
+    axis = ["x", "y", "z"]
+    for i in range(X_reduced.shape[1]):
+        properties[f"Component {i + 1}"] = X_reduced[:, i].tolist()
```

Comment:
you don't need to manually separate the different dimensions, the other functions in chemiscope will do it for you

```suggestion
    properties["features"] = X_reduced
```

Also, this should check that there is not already a `features` key in the properties

Reply:
we can also pick another name than `features` or `components`. I like features, since the function is called `featurize`